src/iadmin.cpp

```diff
@@ -147,6 +149,140 @@ auto get_replica_value(
     return v;
 } // get_replica_value
 
+auto check_that_connected_user_is_rodsadmin(RcComm* _comm) -> bool
+{
+    if (!_comm) {
+        return false;
+    }
+
+    try {
+        namespace adm = irods::experimental::administration;
+
+        const auto user_type =
+            adm::client::type(*_comm, adm::user{_comm->clientUser.userName, _comm->clientUser.rodsZone});
+
+        if (!user_type) {
+            rodsLogError(
+                LOG_ERROR, CAT_INVALID_USER_TYPE, "Could not determine if user has permission to perform operation.");
+            return false;
+        }
+
+        if (*user_type != adm::user_type::rodsadmin) {
+            rodsLogError(LOG_ERROR, CAT_INSUFFICIENT_PRIVILEGE_LEVEL, "Operation requires rodsadmin level privileges.");
+            return false;
+        }
+    }
+    catch (const irods::exception& e) {
+        rodsLogError(LOG_ERROR, e.code(), e.client_display_what());
+        return false;
+    }
+    catch (const std::exception& e) {
+        rodsLogError(LOG_ERROR, SYS_UNKNOWN_ERROR, e.what());
+        return false;
+    }
+
+    return true;
+} // check_that_connected_user_is_rodsadmin
+
+auto print_pam_password_config() -> int
+{
+    if (!check_that_connected_user_is_rodsadmin(Conn)) {
+        return 1;
+    }
+
+    constexpr std::array<const char*, 4> options{
+        "password_length", "no_extend", "password_min_time", "password_max_time"};
+
+    nlohmann::json pam_password_config;
+
+    GridConfigurationInput input{};
+    std::strncpy(input.name_space, "pam_password", sizeof(input.name_space));
+
+    GridConfigurationOutput* output{};
+
+    for (const auto& opt : options) {
+        irods::at_scope_exit free_output{[&output] { std::free(output); }};
+
+        std::strncpy(input.option_name, opt, sizeof(input.option_name));
+
+        if (const auto ec = rc_get_grid_configuration_value(Conn, &input, &output); ec != 0) {
+            char* error_subname = nullptr;
+            const char* error_name = rodsErrorName(ec, &error_subname);
+            const auto msg = fmt::format(
+                "Failed to get pam_password configuration [{}]: [ec={}][msg={}]", opt, error_name, error_subname);
+
+            std::cerr << msg << "\n";
+
+            std::free(error_subname);
+
+            return 1;
+        }
+
+        pam_password_config[opt] = output->option_value;
+    }
+
+    std::cout << pam_password_config.dump(4) << '\n';
+
+    return 0;
+} // print_pam_password_config
+
+auto set_pam_password_config(const char* _option_name, const char* _option_value) -> int
+{
+    if (!check_that_connected_user_is_rodsadmin(Conn)) {
+        return 1;
+    }
+
+    if (!_option_name || !_option_value) {
+        std::cerr << "pam_password configuration option name or value is null.\n";
+        return 1;
+    }
```

Comment:
Consider checking the options first to avoid network traffic.

Reply:
Will do